lib/rules/indent.js

```diff
@@ -1237,8 +1265,24 @@ module.exports = {
             VariableDeclarator(node) {
                 if (node.init) {
                     const equalOperator = sourceCode.getTokenBefore(node.init, astUtils.isNotOpeningParenToken);
+                    const tokenBeforeOperator = sourceCode.getTokenBefore(equalOperator);
                     const tokenAfterOperator = sourceCode.getTokenAfter(equalOperator);
 
+                    /**
+                    * For object and array destructuring that is used across lines
+                    * align the right side elements of the object or array with the equal sign
+                    */
+                    if (isComplementBraceOrBracket(tokenAfterOperator, tokenBeforeOperator)) {
+                        const closingToken = sourceCode.getLastToken(node);
+                        const inBetweenTokens = sourceCode.getTokensBetween(equalOperator, closingToken);
+
+                        inBetweenTokens.forEach(token => {
+                            offsets.matchIndentOf(equalOperator, token);
+                            offsets.ignoreToken(token);
+                        });
+                        offsets.matchIndentOf(tokenBeforeOperator, closingToken);
+                        offsets.ignoreToken(closingToken);
+                    }
                     offsets.ignoreToken(equalOperator);
                     offsets.ignoreToken(tokenAfterOperator);
                     offsets.matchIndentOf(equalOperator, tokenAfterOperator);
```

Comment:
I think the root cause of the problem is that the `=` token's indentation is not set correctly in the `VariableDeclarator` listener. By design, `offsets.ignoreToken` is called on the token, but this is a no-op if the token is not the first of its line. So the `=` token ends up with an offset of 1 from the start of the declarator, and then the value on the right of the equals sign is matched against the indentation of the equals sign. As a result, the rule indents value on the right of the equals sign by 1 from the start of the declarator, when it should be indented by 0.

I think this can be fixed by explicitly specifying the indentation of the `=` token relative to the declarator. Applying the following diff to the original version of the file seems to make everything work correctly and fixes the bug:

```diff
diff --git a/lib/rules/indent.js b/lib/rules/indent.js
index 5a8feee8..4b4355b3 100644
--- a/lib/rules/indent.js
+++ b/lib/rules/indent.js
@@ -1242,6 +1242,7 @@ module.exports = {
                     offsets.ignoreToken(equalOperator);
                     offsets.ignoreToken(tokenAfterOperator);
                     offsets.matchIndentOf(equalOperator, tokenAfterOperator);
+                    offsets.matchIndentOf(sourceCode.getFirstToken(node), equalOperator);
                 }
             },
 

```